app/controllers/queue_items_controller.rb

```diff
@@ -20,6 +20,17 @@ def create
   def destroy
     queue_item = QueueItem.find params[:id]
     current_user.remove_from_queue!(queue_item)
+    current_user.normalize_queue!
+    redirect_to my_queue_path
+  end
+
+  def update
+    begin
+      current_user.update_queue!(params[:queue])
+    rescue
```

Comment:
It is almost never recommended to `rescue` without a specified error type. This will catch everything, including `ActiveRecord::RecordNotValid`, `NoMethodError`, etc. You could be chasing your tail for hours wondering what is going wrong and an all-encompassing `rescue` is suppressing the error that would show you where a simple typo existed.